### PR TITLE
feat: Make builds an implicit part of the deployment

### PR DESCRIPTION
**Pull request by asalkeld**
Allow deployment commands to get gather the stack definition from code.

BREAKING CHANGES: command structure
- remove the build command
- deplyoments now have the "name" passed as a flag
- deployments now accept the handlerGlob

### CODE REVIEW

**Review by tjholm — APPROVED** [review]

File: pkg/build/build.go [hunk 1]
```diff
@@ -73,14 +74,24 @@ func Create(s *stack.Stack, t *target.Target) error {
...
+		}
+		lang := rt.String()
+		_, ok := imagesToBuild[lang]
+		if ok {
+			continue
+		}
 
-	for lang, imageTag := range imagesToBuild {
-		f, err := os.CreateTemp(stackPath, fmt.Sprintf("%s.*.dockerfile", lang))
+		f, err := os.CreateTemp(s.Path(), fmt.Sprintf("%s.*.dockerfile", lang))
```
Comment (tjholm) [inline pkg/build/build.go]: The stackPath contains nitric.yaml, so nitric.yaml is being treated as a directory in this case.

Comment (tjholm) [reply pkg/build/build.go]: see stack/options.go:56

**Review by tjholm — CHANGES_REQUESTED** [review]

File: pkg/cmd/deployment/root.go [hunk 2]
```diff
@@ -19,79 +19,118 @@ package deployment
...
+		}
 		cobra.CheckErr(err)
+
 		p, err := provider.NewProvider(s, t)
 		cobra.CheckErr(err)
+
+		err = build.Create(s, t)
+		cobra.CheckErr(err)
+
 		cobra.CheckErr(p.Apply(args[0]))
```
Comment (tjholm) [inline pkg/cmd/deployment/root.go]: should apply the stackName.

**Review by tjholm — CHANGES_REQUESTED** [review]

File: pkg/codeconfig/codeconfig.go [hunk 3]
```diff
@@ -43,48 +44,55 @@ import (
...
 
-	for _, f := range c.files {
+	for _, f := range c.initialStack.Functions {
 		wg.Add(1)
 
 		// run files in parallel
 		go func(file string) {
 			defer wg.Done()
-			rel, err := filepath.Rel(c.stackPath, file)
+			rel, err := filepath.Rel(c.initialStack.Dir, file)
```
Comment (tjholm) [inline pkg/codeconfig/codeconfig.go]: I think the original glob method we were using was fully qualifying the file paths, this is erroring on attempting to make an unrelated path relative (as it already is relative.)

**Review by tjholm — CHANGES_REQUESTED** [review]

File: pkg/cmd/deployment/root.go [hunk 4]
```diff
@@ -19,79 +19,118 @@ package deployment
...
+
 		p, err := provider.NewProvider(s, t)
 		cobra.CheckErr(err)
-		cobra.CheckErr(p.Delete(args[0]))
+
+		cobra.CheckErr(p.Delete(deploymentName))
 	},
-	Args: cobra.ExactArgs(1),
+	Args: cobra.MinimumNArgs(0),
 }
```
Comment (tjholm) [inline pkg/cmd/deployment/root.go]: Just a thought after using this, we don't really need the handler globs in this case. Just a way to get the stack name, and the name of the deployment right? Doesn't need to be done in this PR and happy to discuss it as well.

Comment (asalkeld) [reply pkg/cmd/deployment/root.go]: done

**Review by tjholm — APPROVED** [review]

**Pull request merged**